edgelet/edgelet-core/src/network.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use crate::DEFAULT_NETWORKID;
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Network {
+    name: String,
+
+    #[serde(rename = "ipv6", skip_serializing_if = "Option::is_none")]
+    ipv6: Option<bool>,
+
+    #[serde(rename = "ipam", skip_serializing_if = "Option::is_none")]
+    ipam: Option<Vec<Ipam>>,
+}
+
+impl Network {
+    pub fn new(name: String) -> Self {
+        Network {
+            name,
+            ipv6: None,
+            ipam: None,
+        }
+    }
+
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn with_name(mut self, name: String) -> Self {
+        self.name = name;
+        self
+    }
+
+    pub fn ipv6(&self) -> Option<&bool> {
+        self.ipv6.as_ref()
+    }
+
+    pub fn with_ipv6(mut self, ipv6: Option<bool>) -> Self {
+        self.ipv6 = ipv6;
+        self
+    }
+
+    pub fn ipam(&self) -> Option<&Vec<Ipam>> {
+        self.ipam.as_ref().map(AsRef::as_ref)
+    }
+
+    pub fn with_ipam(mut self, ipam: Option<Vec<Ipam>>) -> Self {
+        self.ipam = ipam;
+        self
+    }
+}
+
+#[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct Ipam {
+    #[serde(rename = "gateway", skip_serializing_if = "Option::is_none")]
+    gateway: Option<String>,
+
+    #[serde(rename = "subnet", skip_serializing_if = "Option::is_none")]
+    subnet: Option<String>,
+
+    #[serde(rename = "ip_range", skip_serializing_if = "Option::is_none")]
+    ip_range: Option<String>,
+}
+
+impl Ipam {
+    pub fn gateway(&self) -> Option<&str> {
+        self.gateway.as_ref().map(AsRef::as_ref)
+    }
+
+    pub fn with_gateway(mut self, gateway: String) -> Self {
+        self.gateway = Some(gateway);
+        self
+    }
+
+    pub fn subnet(&self) -> Option<&str> {
+        self.subnet.as_ref().map(AsRef::as_ref)
+    }
+
+    pub fn with_subnet(mut self, subnet: String) -> Self {
+        self.subnet = Some(subnet);
+        self
+    }
+
+    pub fn ip_range(&self) -> Option<&str> {
+        self.ip_range.as_ref().map(AsRef::as_ref)
+    }
+
+    pub fn with_ip_range(mut self, ip_range: String) -> Self {
+        self.ip_range = Some(ip_range);
+        self
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum MobyNetwork {
+    Network(Network),
+    Name(String),
+}
+
+impl MobyNetwork {
+    pub fn name(&self) -> &str {
+        match self {
+            MobyNetwork::Name(name) => {
+                if name.is_empty() {
```

Comment:
Can you make it `Name(Option<String>)` so that you can use `name.unwrap_or(DEFAULT_NETWORK_ID)` instead of empty string sentinel? (But make sure that `serde(untagged)` can still handle it.)

Reply:
As discussed offline, the default embedded config would make it impossible to get `None` here anyway, so leave this as-is.